server/src/test/java/org/elasticsearch/common/xcontent/support/XContentMapValuesTests.java

```diff
@@ -48,10 +56,235 @@
 
 public class XContentMapValuesTests extends AbstractFilteringTestCase {
 
+    private static final String NO_INCLUDE = null;
```

Comment:
lots of tests, for all combos. The previous tests were unreadable, every test basically asserts against an expected map, making it easy to see exactly what is missing/extra.